server/src/main/java/org/candlepin/resource/SubscriptionResource.java

```diff
@@ -63,34 +65,37 @@
 @Path("/subscriptions")
 @Api(value = "subscriptions", authorizations = { @Authorization("basic") })
 @Consumes(MediaType.APPLICATION_JSON)
-public class SubscriptionResource {
+public class SubscriptionResource implements SubscriptionsApi {
     private static Logger log = LoggerFactory.getLogger(SubscriptionResource.class);
 
-    private SubscriptionServiceAdapter subService;
-    private ConsumerCurator consumerCurator;
-    private PoolManager poolManager;
-
-    private I18n i18n;
+    private final SubscriptionServiceAdapter subService;
+    private final ConsumerCurator consumerCurator;
+    private final PoolManager poolManager;
+    private final I18n i18n;
+    private final ModelTranslator translator;
 
     @Inject
     public SubscriptionResource(SubscriptionServiceAdapter subService,
-        ConsumerCurator consumerCurator, PoolManager poolManager, I18n i18n) {
-
-        this.subService = subService;
-        this.consumerCurator = consumerCurator;
-        this.poolManager = poolManager;
-
-        this.i18n = i18n;
+        ConsumerCurator consumerCurator, PoolManager poolManager, I18n i18n,
+        ModelTranslator translator) {
+
+        this.subService = Objects.requireNonNull(subService);
+        this.consumerCurator = Objects.requireNonNull(consumerCurator);
+        this.poolManager = Objects.requireNonNull(poolManager);
+        this.i18n = Objects.requireNonNull(i18n);
+        this.translator = Objects.requireNonNull(translator);
     }
 
     @ApiOperation(notes = "Retrieves a list of Subscriptions", value = "getSubscriptions")
     @GET
     @Produces(MediaType.APPLICATION_JSON)
```

Comment:
We're moving this resource to the spec file, which means all these annotations should be removed from all endpoints here (since they will be auto-generated in SubscriptionApi)

Reply:
Done.